satellite/satellitedb/migrate.go

```diff
@@ -0,0 +1,330 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"database/sql"
+	"strconv"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/migrate"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/satellite/console"
+)
+
+// ErrMigrate is for tracking migration errors
+var ErrMigrate = errs.Class("migrate")
+
+// CreateTables is a method for creating all tables for database
+func (db *DB) CreateTables() error {
+	switch db.driver {
+	case "postgres":
+		migration := db.PostgresMigration()
+		return migration.Run(db.log.Named("migrate"), db.db)
+	default:
+		return migrate.Create("database", db.db)
+	}
+}
+
+// PostgresMigration returns steps needed for migrating postgres database.
+func (db *DB) PostgresMigration() *migrate.Migration {
+	return &migrate.Migration{
+		Table: "versions",
+		Steps: []*migrate.Step{
```

Comment:
This file will grow very large.  Should we start a pattern where different steps are stored as individual files?

Reply:
The problem is that not all the steps are SQL... also that would mean we need to start embedding the sql files anyways. As for largeness, I think satellites can afford to drop older migrations and just setup a newer ones without problems.

Reply:
I didn't mean storing the SQL as test files, simply:

return &migrate.Migration{
		Table: "versions",
		Steps: [
		  Step1,
		  Step2,
		  ...
		]
---
with separate files for:
Step1.go
Step2.go
...

Reply:
Ah, sure, might make sense... we'll see how the whole thing evolves. `migrate_v1.go`, `migrate_v2to4.go` etc.

Reply:
Why should anyone change an existing migration step or do you mean preventing that? Whether separate files or not, just seems weird putting a single statement in a separate file; but I'm not against the idea either. What do you exactly mean by applying step manually?

`migrate_v2to4`, nah that was just so we don't have a bunch of single line files... but can be separate for consistency.

Reply:
Yes, I meant preventing changes to existing steps.

Applying a step manually might look like `psql -1f migrations/20190214-add-and-index-customers-table.sql satellitedb`.